src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *           type: string
+ *         description: The unique identifier of the post to be updated.
+ *     requestBody:
+ *       description: The fields of the post to be updated.
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               title:
+ *                 type: string
+ *                 description: The new title of the post.
+ *               content:
+ *                 type: string
+ *                 description: The new content of the post.
+ *               price:
+ *                 type: number
+ *                 description: The new price of the post.
+ *               category_url:
+ *                 type: string
+ *                 description: The new category URL of the post.
+ *               [other fields]:
+ *                 type: string
+ *                 description: Other fields that may be updated.
+ *     responses:
+ *       200:
+ *         description: Success message with updated post data.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 data:
+ *                   type: object
+ *                   description: The result of the update operation.
+ *                   properties:
+ *                     modifiedCount:
+ *                       type: integer
+ *                       description: Number of documents modified.
+ *                 msg:
+ *                   type: string
+ *                   description: Status message indicating that the post was edited.
+ *       400:
+ *         description: Bad request, indicating the update operation was not successful or if the user is not authorized to update the post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 data:
+ *                   type: object
+ *                   description: Details of the failed update operation.
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating why the post could not be updated.
+ *                 err:
+ *                   type: object
+ *                   description: Details of any error encountered.
+ *       401:
+ *         description: Unauthorized error if the user does not have permission to update the post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating unauthorized access.
+ */
 router.put('/:postID', auth, updatePost);
+
+/**
+ * @swagger
+ * /posts/changeRange/{postID}:
+ *   put:
+ *     tags:
+ *       - Posts operations
+ *     summary: Change the range of a post , Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"